week06/completed_product_test.go

```diff
@@ -0,0 +1,71 @@
+package week06
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestCompletedProduct_IsValid(t *testing.T) {
+
+	t.Parallel()
+
+	//Build a product of 1 Quantity for 1 Employee
+	myProduct := Product{Quantity: 1}
+	myProduct.Build(1)
+
+	//Build a product of 0 Quantity for 1 Employee
+	myProductNoQuantity := Product{}
+	myProductNoQuantity.Build(1)
+
+	//Different Complete Products for different scenarios
+	emptyCompleteProduct := CompletedProduct{}
+	completeProductNoEmployee := CompletedProduct{Product: myProduct, Employee: Employee(0)}
+	completeProductNotBuild := CompletedProduct{Product: Product{Quantity: 1}, Employee: Employee(1)}
+	completeProductNoQuantity := CompletedProduct{Product: myProductNoQuantity, Employee: Employee(1)}
+	completeProductFull := CompletedProduct{Product: myProduct, Employee: Employee(1)}
```

Comment:
These variables are far too long for idiomatic Go. Try shortening them.

In addition, you are testing too much. The `CompletedProduct` validates against `Product.IsBuilt` so as long as that returns `false`, you're good. You don't need to test additional varieties of that failure here, such as an invalid quantity on the `Product`. Those scenarios belong with the unit test for `Product.IsBuilt`.

```suggestion
	prod := Product{Quantity: 1}
	prod.Build(1)
	
	e := Employee(1)

	//Different Complete Products for different scenarios
	
	empty := CompletedProduct{}
	noEmployee := CompletedProduct{Product: prod}
	notBuilt := CompletedProduct{Product: Product{Quantity: 1}, Employee: e}
	good := CompletedProduct{Product: myProduct, Employee: e}
```